### PR TITLE
Metrics server upgrade

### LINKED ISSUE #665: Upgrade metrics-server to v0.5.1
**What Chart should be upgraded?**
upstream / metrics-server: [metrics-server/metrics-server](https://github.com/kubernetes-sigs/metrics-server/tree/master/charts/metrics-server)
v0.3.6/v2.10.0 (2019/10/14) -> v0.5.1/v3.6.0 (2021/10/04)

**Why should this Chart be upgraded?**

- [ ] We need to perform a security upgrade.
- [ ] We need it for a new feature: *which feature (link to blocked issue)*.
- [x] We need to keep up to prevent future issues.
- [ ] Other: *Write the reason here*

Need to upgrade this to support K8s v1.22 going forward.

Lots of changes.

**Acceptance criteria**

- I checked the migration of the new Chart:
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).
- [ ] I tested the functionality provided by the new Chart (e.g., Auth flow, Grafana dashboards, etc.)

### PR DESCRIPTION
**Pull request by crssnd**
**What this PR does / why we need it**: to upgrade the metrics-server chart

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #665 

**Special notes for reviewer**:

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [x] is completely transparent, will not impact the workload in any way.
    - [x] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [x] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: helmfile/values/metrics-server.yaml.gotmpl [hunk 1]
```diff
@@ -2,9 +2,22 @@ rbac:
   create: true
   pspEnabled: true
 
+# added just to keep only InternalIP as kubelet preferred address
+defaultArgs:
+  - --cert-dir=/tmp
+  - --kubelet-preferred-address-types=InternalIP
+  - --kubelet-use-node-status-port
+  - --metric-resolution=15s
```
Comment (viktor-f) [inline helmfile/values/metrics-server.yaml.gotmpl:10]: I think we scrape most (or all) endpoints every 30 seconds. Any reason why this is 15s?

**Review by viktor-f — APPROVED** [review]
LGTM
Just remove the prometheus commits before merging.

**Review by Xartos — APPROVED** [review]
LGTM

File: helmfile/values/metrics-server.yaml.gotmpl (hunk 1 above)
Comment (crssnd) [reply helmfile/values/metrics-server.yaml.gotmpl:10]: I just kept the default value.
I will change it to 30s, but this is not he same as the Prometheus scrape interval. This value is used by the metrics server in the autoscalling process.

Comment (viktor-f) [reply helmfile/values/metrics-server.yaml.gotmpl:10]: Ok, then I think this is ok to keep at 15s if you want. You can choose.

**Pull request merged**